include/eve/algo/zip.hpp

```diff
@@ -0,0 +1,126 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/algo/as_range.hpp>
+#include <eve/algo/preprocess_range.hpp>
+#include <eve/algo/traits.hpp>
+#include <eve/algo/zip_iterator.hpp>
+
+#include <eve/algo/detail/preprocess_zip_range.hpp>
+
+#include <eve/detail/kumi.hpp>
+
+namespace eve::algo
+{
+  template <typename ...Rngs>
+  struct zip_range
+  {
+    kumi::tuple<Rngs...> ranges;
+
+    auto begin() const
+    {
+      return zip_iterator(kumi::map([](auto r) { return r.begin(); }, ranges));
+    }
+
+    auto end() const
+    {
+      return zip_iterator(kumi::map([](auto r) { return r.end(); }, ranges));
+    }
+
+    template <typename Traits>
+    friend auto tagged_dispatch(preprocess_range_, Traits traits, zip_range self)
+    {
+      return preprocess_zip_range(traits, self.ranges);
+    }
+  };
+
+  template <typename ...Rngs>
+  zip_range(kumi::tuple<Rngs...>) -> zip_range<Rngs...>;
+
+  namespace detail
+  {
+    // FIX-874: we should do this proper
```

Comment:
DId you check what is in `eve/concept/range.hpp` ?

Reply:
That's not quite what I want.

This should be `eve::algo::light_range`.

For example `eve::aligned_ptr` is not an iterator but it should be ok for `light_range`